clothing_app/clothing_app/spiders/ackermans.py

```diff
@@ -0,0 +1,242 @@
+import copy
+import json
+from urllib.parse import urljoin
+
+import scrapy
+
+from ..items import ProductItem, SizeItem
+
+
+class AckermansSpider(scrapy.Spider):
+    name = "ackermans"
+
+    def start_requests(self):
+        url = "https://www.ackermans.co.za/"
+        yield scrapy.Request(url, self.parse)
+
+    navigation_api = "https://www.ackermans.co.za/api/startup-data.json"
+    products_api = "https://www.ackermans.co.za/graphql?operationName=products_listAndAggregations"
+    product_data_api = "https://www.ackermans.co.za/graphql?operationName=products_pdp"
+    product_base_url = "https://www.ackermans.co.za/products/"
+    product_image_base_url = "https://www.ackermans.co.za/imagekit/prod-ack-product-images/"
+
+
+    page_size = 20
+
+    products_headers = {
+        "Content-Type": "application/json",
+        "accept": "application/json, text/plain, */*",
+    }
+
+    def parse(self, response):
```

Comment:
We are not extracting any data in this `parse` function. We can directly add navigation_api URL in start_urls.
Alternatively, we can implement our own `start_requests`.